### ISSUE TITLE
Dots in field names are replaced silently

### ISSUE THREAD

**Issue by mpfz0r** [Graylog2/graylog2-server#13043]
The fact that we don't support dots in field names is nothing new.
This code has been there since ages, silently replacing dots with underscores once a Message gets written to ES/OS:

https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/plugin/Message.java#L404-L406

What is problematic, is that we neither have documentation for this behavior, nor do we log a warning if we replace dots.

The fact that field renaming happens very late in the processing can lead to some confusion when writing pipeline rules or extractors.
The ingested Message will show up with underscore fields in the search, but within processing you need to work with dots.

## Possible solutions
 - Add documentation
 - Show a system notification if we detect that changing field names was needed. (This needs to be done efficiently to not cause a performance degradation)
 - ~Change the field names directly when assigning them~ (This will break existing rules and might confuse users who are already accustomed to the existing behavior)

Refs:
https://github.com/Graylog2/graylog2-server/issues/12990
https://github.com/Graylog2/graylog2-server/issues/6588
https://github.com/Graylog2/graylog2-server/pull/5983

https://github.com/Graylog2/graylog2-server/issues/4583
https://github.com/elastic/elasticsearch/issues/15951

**Comment by coffee-squirrel** [comment]
Mentioned on the case, but being able to `debug(to_string($message))` (or equivalent) would've helped greatly in figuring out what was going on.

**Comment by mpfz0r** [comment]
@coffee-squirrel Like this? https://github.com/Graylog2/graylog2-server/pull/13178

**Comment by mpfz0r** [comment]
@coffee-squirrel Yeah, makes sense. I didn't do that in the beginning, because it required some changes to the rule parser.
But it should work now.

**Comment by OzzyKampha** [comment]
Can this get fixed. Since the new versions of elastic have . As standard now

**Comment by pandel** [comment]
Will this ever be solved? The [issue](https://github.com/Graylog2/graylog2-server/issues/4583) was created 5+ years ago and it seems, as if the whole thing hasn't been addressed yet, even though there are many people, which seem to have problems with this "dots aren't allowed" thing.

**Comment by FlatCodeIq** [comment]
Hello,

We need this case to be addressed ASAP, Wazuh Dashboard is unable to read the logs due to the separator "_" instead of ".".

**Comment by janheise** [comment]
@pandel @FlatCodeIq What exactly is the problem with Wazuh Dashboard? I'm not familiar with that product. 
The `.` will be addressed at some point - but I can not give an estimate as it's a major change (e.g. a `.` in a key creates a nested object) and we have to keep backwards compatibility in mind. But we're discussing it.

**Comment by pandel** [comment]
@janheise [Wazuh](https://wazuh.com) is a very decent and popular open source SIEM solution. Every data element is organized via  `.` structured field names. The main dashboards, which are provided by the developers, are not able to find the data fields when all `.` are replaced with `_`, that simple.

Wazuh itself normally uses filebeat, but if you replace filebeat with graylog (i.e. to use [CoPilot](https://github.com/socfortress/CoPilot) with all of its additional features), you loose capabilities which the Wazuh dashboards normally provide...

**Comment by DionysiusInc** [comment]
@janheise Any update on this issue? It's not possible to integrate GL into the Wazuh stack because the replacement of `.` to `_` breaks all dashboards and visualizations in the Wazuh dashboard.  Which causes us to be stuck with Filebeat which is far less superior.

**Comment by pandel** [comment]
And again: is there any update on this? A solution to this would be highly appreciated because still, it is really annoying that Graylog is still replacing the `.` and breaking things by that.

**Comment by Som-gif** [comment]
Hello everyone, any news about that ?
Message from 2022, can we maybe get a final answer :)

Thanks
